src/Payment.ts

```diff
@@ -92,24 +92,36 @@ export class Payment {
    * @param options.feePayer Either `sender` or `receiver`. Specifies who pays network fees.
    * @param options.extraData Extra data that will appear in the Transfer event when successful.
    * @param options.paymentRequestId Payment request identifier that gets encoded to the extra data.
+   * @param options.addMessageId Whether a message id should be added to the payment, default to true.
    */
   public async prepare(
     networkAddress: string,
     receiverAddress: string,
     value: number | string,
-    options: PaymentOptions = {}
+    options: PaymentOptions = { addMessageId: true }
   ): Promise<PaymentTxObject> {
     const {
       gasPrice,
       gasLimit,
       networkDecimals,
       extraData,
-      paymentRequestId
+      paymentRequestId,
+      addMessageId
     } = options
     const decimals = await this.currencyNetwork.getDecimals(networkAddress, {
       networkDecimals
     })
-    const encodedExtraData: string = encode({ extraData, paymentRequestId })
+    let messageId: string
+    if (addMessageId) {
+      // 19 decimals fit in 64 bits
```

Comment:
Sorry I feel stupid, but I fail to do this calculation/get the same result. Could you maybe extend the comment?

Reply:
`2**64 = 1.8446744e+19` so `1e+19 < 2**64`. We could also arbitrarily decide to use more, which is why I didn't want to make the comment too complex, I don't want to make it sound like a hard requirement.